external_plugins/custom_tokens/custom_tokens_plugin.cpp

```diff
@@ -0,0 +1,367 @@
+
+            plugin_.save_token_operation(from, amount.symbol, note_);
+            plugin_.save_token_operation(*op.recipients.begin(), amount.symbol, note_);
+
+         } else if( tmp[ "action" ].as<string>() == std::string("burn_token")) {
+
+            auto amount = tmp[ "amount" ].as<asset>();
+
+            plugin_.check_if_paused(amount.symbol, note_.trx_id);
+
+            plugin_.remove_account_tokens(from, amount.symbol, amount.amount.value, note_.trx_id);
+            plugin_.save_token_operation(from, amount.symbol, note_);
+
+            auto token = plugin_.db_->find<custom_token_object, by_token_symbol>(amount.symbol);
+
+            if( token ) {
+               plugin_.db_->modify(*token, [ & ](custom_token_object &to) {
+                    to.burned += amount.amount.value;
+               });
+            }
+
+         } else if( tmp[ "action" ].as<string>() == std::string("issue_token")) {
+
+            auto token_symbol = asset_symbol_type::from_string(tmp[ "token_symbol" ].as<string>());
+            plugin_.check_if_paused(token_symbol, note_.trx_id);
+
+            auto additional_amount = tmp[ "additional_amount" ].as<uint64_t>();
+            auto token = plugin_.db_->find<custom_token_object, by_token_symbol>(token_symbol);
+
+            FC_ASSERT(token, "${s}", ("s", plugin_.save_token_error(note_.trx_id, "No such a token!")));
+
+            if( token->owner_name == from &&
+                token->max_supply > token->total_supply + additional_amount ) {
+               plugin_.db_->modify(*token, [ & ](custom_token_object &to) {
+                    to.total_supply += additional_amount;
+               });
+
+               plugin_.add_account_tokens(from, token_symbol, additional_amount);
+               plugin_.save_token_operation(from, token_symbol, note_);
+
+            }
+
+         } else {
+            FC_ASSERT(false, "${s}",
+                      ("s", plugin_.save_token_error(note_.trx_id, "Unknown action for custom token!")));
+         }
+      }
+   }
+};
+
+void custom_tokens_plugin_impl::post_operation(const operation_notification &note) {
+   note.op.visit(post_operation_visitor(*this, note));
+}
+
+void custom_tokens_plugin_impl::check_if_paused(asset_symbol_type token_symbol, const transaction_id_type &tx_id) {
+
+   auto token = db_->find<custom_token_object, by_token_symbol>(token_symbol);
+   FC_ASSERT(token, "${s}", ("s", save_token_error(tx_id, "No such a token!", token_symbol)));
+   FC_ASSERT(!token->paused, "${s}",
+             ("s", save_token_error(tx_id, "Can to do any action with token, because it is paused!", token_symbol)));
+}
+
+void custom_tokens_plugin_impl::add_account_tokens(const account_name_type &name, asset_symbol_type token_symbol,
+                                                   uint64_t amount) {
+   auto account = db_->find<custom_token_account_object, by_token_and_account>(
```

Comment:
shouldn't it be called rather "balance" or something similar? "account" is quite confusing...

Reply:
fixed